src/net/query.c

```diff
@@ -13,6 +13,10 @@
 
 #include "zenoh-pico/net/query.h"
 
+void _z_query_clear(_z_query_t *q) {
```

Comment:
This is the final destructor (called when refcount hits 0) for the query, right?

Then this is likely where all the memory freeing and `REPLY_FINAL` triggering logic should be.